### PR TITLE
[WX-965] quote() and squote() engine functions.

### PR DESCRIPTION
**Pull request by rsaperst**
Implemented [quote()](https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#-arraystring-quotearrayp) and [squote()](https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#-arraystring-squotearrayp) engine functions, which are new with WDL 1.1.

As a new WDL 1.1 feature, it was implemented for `Biscayne` and `Cascades`, but not earlier versions (e.g. `draft-2`, `draft-3`).

### CODE REVIEW

File: wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/files/BiscayneFileEvaluators.scala [hunk 1]
```diff
@@ -17,6 +28,10 @@ object BiscayneFileEvaluators {
 
   implicit val sepFunctionEvaluator: FileEvaluator[Sep] = twoParameterFunctionPassthroughFileEvaluator[Sep]
   implicit val suffixFunctionEvaluator: FileEvaluator[Suffix] = twoParameterFunctionPassthroughFileEvaluator[Suffix]
+  implicit val quoteFunctionEvaluator: FileEvaluator[Quote] =
+    EngineFunctionEvaluators.singleParameterPassthroughFileEvaluator
+  implicit val sQuoteFunctionEvaluator: FileEvaluator[SQuote] =
+    EngineFunctionEvaluators.singleParameterPassthroughFileEvaluator
```
Comment (jgainerdewar) [inline wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/files/BiscayneFileEvaluators.scala:34]: Nitpick for consistency - can you update the import to handle `singleParameterPassthroughFileEvaluator` the same way as `twoParameterFunctionPassthroughFileEvaluator` so they can be referenced the same way here?

File: wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala [hunk 2]
```diff
@@ -235,4 +235,46 @@ object BiscayneValueEvaluators {
...
+                               inputs: Map[String, WomValue],
+                               ioFunctionSet: IoFunctionSet,
+                               forCommandInstantiationOptions: Option[ForCommandInstantiationOptions]
+    )(implicit expressionValueEvaluator: ValueEvaluator[ExpressionElement]): ErrorOr[EvaluatedValue[WomArray]] =
+      processValidatedSingleValue[WomArray, WomArray](
+        expressionValueEvaluator.evaluateValue(a.param, inputs, ioFunctionSet, forCommandInstantiationOptions)(
+          expressionValueEvaluator
+        )
+      ) { arr =>
+        EvaluatedValue(WomArray(arr.value.map(v => WomString("\"" + v.valueString + "\""))), Seq.empty).validNel
```
Comment (jgainerdewar) [inline wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala:256]: Do you think we need to worry about escaping the value? The spec doesn't say anything about that, but as a user I would expect any quotes that already exist in the string to get escaped.

File: wdl/transforms/biscayne/src/test/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluatorSpec.scala [hunk 3]
```diff
@@ -217,4 +217,128 @@ class BiscayneValueEvaluatorSpec extends AnyFlatSpec with CromwellTimeoutSpec wi
       e.evaluateValue(Map.empty, NoIoFunctionSet, None) shouldBeValid EvaluatedValue(expectedArray, Seq.empty)
     }
   }
+
```
Comment (jgainerdewar) [inline wdl/transforms/biscayne/src/test/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluatorSpec.scala:220]: Thorough tests!

Comment (jgainerdewar) [reply wdl/transforms/biscayne/src/test/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluatorSpec.scala:220]: As discussed in mob time, to make them even more thorough... we should test with every primitive data type.

https://github.com/openwdl/wdl/blob/wdl-1.1/SPEC.md#primitive-types

**Comment by aednichols** [comment]
Holding off a thumb until we everything compiling and passing.

Pro tip, you can force full compilation of tests without running all of them by issuing `Test / compile` at the SBT prompt.

File: wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/types/BiscayneTypeEvaluators.scala [hunk 4]
```diff
@@ -121,6 +121,32 @@ object BiscayneTypeEvaluators {
       ) mapN { (_, _) => WomArrayType(WomStringType) }
   }
 
+  implicit val quoteFunctionEvaluator: TypeEvaluator[Quote] = new TypeEvaluator[Quote] {
+    override def evaluateType(a: Quote, linkedValues: Map[UnlinkedConsumedValueHook, GeneratedValueHandle])(implicit
+      expressionTypeEvaluator: TypeEvaluator[ExpressionElement]
+    ): ErrorOr[WomType] =
+      validateParamType(a.param, linkedValues, WomArrayType(WomAnyType)) flatMap {
+        case WomArrayType(_: WomPrimitiveType) => WomArrayType(WomStringType).validNel
```
Comment (THWiseman) [inline wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/types/BiscayneTypeEvaluators.scala:129]: If I recall, an empty array becomes `WomArrayType(WomNothingType)`, which doesn't seem to match any of the valid cases, despite it being valid input. Should that be handled here?

**Review by jgainerdewar — APPROVED** [review]
Nice work!! Assuming tests pass with the latest changes from `develop`, looks good to me!

**Pull request merged**